File: DesignPatternsExercises/java/TemplateMethodDesignPattern/Exercise/AirTravel.java
public class AirTravel extends Travel {
    @Override
    public void setDates() {
        System.out.println("Setting dates...");
        this.startDate = "07/04/2019";
        this.endDate = "08/10/2019";
    }

    @Override
    public void setDestination() {
        System.out.println("Setting destination...");
        this.destination = "Tokyo, Japan";
    }

    @Override
    void setMeal() {
        System.out.println("Selecting meal...");
        this.meal = "Asian Vegetarian";
    }

    @Override
    void usePrivateVehicle() {}

    @Override
    void bookTicket() {
        System.out.println("Booking ticket...");
        booked = true;
    }

    @Override
    void displayTravelDetails() {
        System.out.println("Air Travel:");
        if (booked) {
            __________(1)__________.displayTravelDetails();
            if (meal != null) {
                System.out.println("        Meal selected: " + meal);
            }
        } else {
            System.out.println("    Transaction cancelled");
        }
        System.out.println();
    }

    @Override
    boolean isMealAvailable() {
        return true;
    }
}

File: DesignPatternsExercises/ruby/template_method_design_pattern/solution/air_travel.rb
require_relative 'travel'

# class AirTravel < __________(1)__________
class AirTravel < Travel
  def dates
    puts 'Setting dates...'
    @start_date = '07/04/2019'
    @end_date = '08/10/2019'
  end

  def destination
    puts 'Setting destination...'
    @destination = 'Tokyo, Japan'
  end

  def meal
    puts 'Selecting meal...'
    @meal = 'Asian Vegetarian'
  end

  def use_private_vehicle
  end

  def book_ticket
    puts 'Booking ticket...'
    @booked = true
  end

  def display_travel_details
    print 'Air Travel:'
    if @booked
      # __________(2)__________
      super
      puts "    Meal selected: #{@meal}" unless @meal.nil?
    else
      puts ' Transaction cancelled'
    end
    puts
  end

  def meal_available?
    true
  end
end
